app/controllers/Tasks_controller.rb

```diff
@@ -0,0 +1,86 @@
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all.order(:name)
+  end
+
+  def show
+    task_id = params[:id]
+    @task = Task.find_by(id: task_id)
+
+    if @task.nil?
+      redirect_to action: "index"
+    end
+  end
+
+  def new
+    @task = Task.new(complete: false, complete_date: nil)
+  end
+
+  def create
+    task = Task.new(
+      name: params["task"]["name"],
+      description: params["task"]["description"],
+      complete: params["task"]["complete"],
+      complete_date: params["task"]["complete_date"],
+    )
```

Comment:
I'd probably refactor this to use the strong params, so `Task.new(task_params)` here instead of this hash